apps/studio/src/features/editing-experience/LayoutSelection.tsx

```diff
@@ -0,0 +1,243 @@
+                          borderRadius="6px" // due to this being smaller than the outer Box
+                        >
+                          <VStack
+                            w="100%"
+                            h="100%"
+                            justify="center"
+                            gap="0.25rem"
+                            opacity="1"
+                            color="base.content.brand"
+                          >
+                            <Icon as={BiShow} />
+                            <Text textStyle="caption-1">Click to preview</Text>
+                          </VStack>
+                        </Box>
+                      )}
+                    </Box>
+                    <HStack w="100%" justifyContent="space-between">
+                      <Text
+                        className="layout-title"
+                        textAlign="left"
+                        textStyle="h6"
+                        textColor={
+                          selectedLayout.layoutTypename === e.layoutTypename
+                            ? 'base.content.brand'
+                            : 'base.content.strong'
+                        }
+                        mb="0.5rem"
+                      >
+                        {e.layoutDisplayName} layout
+                      </Text>
+                    </HStack>
+                    <Text
+                      textAlign="left"
+                      textStyle="body-1"
+                      textColor="base.content.default"
+                    >
+                      {e.layoutDescription}
+                    </Text>
+                  </VStack>
+                )
+              })}
+            </VStack>
+          </GridItem>
+          <GridItem
+            w="100%"
+            colSpan={3}
+            bg="grey.100"
+            p="2rem 2rem 0 2rem"
+            overflow="auto"
+          >
+            <Box
+              borderRadius="8px 8px 0px 0px"
+              w="100%"
+              minH="100%"
+              boxShadow="0 0 20px 0 rgba(104,104,104,0.3)"
+              position="relative"
+            >
+              <HStack
+                borderRadius="8px 8px 0px 0px"
+                w="100%"
+                bgColor="slate.200"
+                textColor="white"
+                justifyContent="center"
+                gap="0"
+                padding="0.5rem 0.75rem"
+              >
+                <Text textStyle="caption-2">
+                  {`You're previewing the`}&nbsp;
+                </Text>
+                <Text textStyle="caption-1">
+                  {selectedLayout.layoutDisplayName}
+                  {` Layout`}
+                </Text>
+              </HStack>
+              <Preview schema={placeholderContent} />
+
+              <Box
+                position="absolute"
+                top="0"
+                left="0"
+                w="100%"
+                h="100%"
+                zIndex="10"
```

Comment:
using `zIndex: number` is scary because it might be positioned behind/over things unexpectedly. perhaps we could do better by using the z-index export from chakra and using a property there.

alternatively, we could consider using `portals` here - it sounds like a perfect use case! see [here](https://v2.chakra-ui.com/community/recipes/z-index#whats-wrong-with-z-index) for more

Reply:
Realised that by placing the div after the preview component, it is rendered correctly above the preview on the z-axis. Is there a need to still specify a z-index export from chakra in this case?
Also, looked into portals, which seems plausible for this, problem is that the div's width and height will have to be rethought(as well as the anchoring using position=absolute), since it is now placed at a different place in the DOM tree, with different parent.

Reply:
in this case, no! can i trouble you to remove zindex